### ISSUE TITLE
cli: generate temporary password based on current password requirements setting

### ISSUE THREAD

**Issue by kimskimchi** [infrahq/infra#2650]
## Is your feature request related to a problem? Please describe.

**Current behaviours:** 
1. It will error out if the user changes the settings and the generated password does not meet it. 
2. when a new user is created, password is generated on the server side.
3. when admin resets a user's password, password is generated on the client-side. The temporary password generated by the cli in: 
   ```
   cmd/users.go/updateUser: 	
   tmpPassword, err := generate.CryptoRandom(12, generate.CharsetPassword)
    ```

**To do:** 
Move #3 client-side password logic to server-side. So, UpdatePassword API endpoint should:
a) generate if an admin is trying to update another user's password, so no password is passed in the request
b) all other cases requires the password to be in the request

~~Blocked by: #2648~~
Blocked by: #2797

**Comment by kimskimchi** [comment]
No longer needed, will switch from temporary passwords to links, even without email servers.
